Models/AbstractProjectModels/AbstractUserModel.php

```diff
@@ -0,0 +1,163 @@
+<?php
+declare(strict_types=1);
+
+namespace Models\AbstractProjectModels;
+
+use Models\AbstractProjectModels\Message\User\AbstractMsgModel;
+use Models\ProjectModels\Sql\MySql\MySqlDbWorkModel;
+
+abstract class AbstractUserModel extends AbstractDefaultModel
+{
+    protected MySqlDbWorkModel $db;
+    protected AbstractMsgModel $msgModel;
+    protected array $user = [];
+
+    public function __construct(AbstractMsgModel $msgModel)
+    {
+        parent::__construct();
+        $this->msgModel = $msgModel;
+        $this->db = new MySqlDbWorkModel();
+    }
+
+    /**
+     * @param string $tableName
+     * @param array $data
+     * @return bool
+     */
+    public function newUser(string $tableName, array $data): bool
+    {
+        if ($this->userExist($tableName, $data['login'], $data['email'], $data['phone'])) {
+            $this->msgModel->setMsg($this->msgModel->getMessage('registration','user_exist'));
+            return false;
+        }
+
+        $data['pass'] = password_hash($data['pass'], PASSWORD_BCRYPT);
+        if ($this->fileInfo->isImageSent()) {
+            $data['image'] = $this->moveUploadFile($tableName);
+        }
+
+        if (!$this->db->insertData($tableName, $data)) {
+            $this->msgModel->setMsg($this->msgModel->getMessage('registration', 'project_mistake'));
+            return false;
+        } else {
+            $data = $this->setDbSpecialFieldsData($data);
+            $this->setSessionData($data);
+            $this->msgModel->setMsg($this->msgModel->getMessage('registration', 'user_reg_success'));
+            return true;
+        }
+    }
+
+    protected function setDbSpecialFieldsData(array $data): array
+    {
+        $specialData['is_active'] = 1;
+        return array_merge($data, $specialData);
```

Comment:
You don't need to use array_merge in order to add only one additional element in array. It is redundant. Please use array expand like 
`$data['is_active'] = 1;`